hardware_interface/src/resource_manager.cpp

```diff
@@ -25,16 +25,46 @@
 #include "hardware_interface/actuator.hpp"
 #include "hardware_interface/actuator_interface.hpp"
 #include "hardware_interface/component_parser.hpp"
+#include "hardware_interface/hardware_component_info.hpp"
 #include "hardware_interface/sensor.hpp"
 #include "hardware_interface/sensor_interface.hpp"
 #include "hardware_interface/system.hpp"
 #include "hardware_interface/system_interface.hpp"
+#include "hardware_interface/types/hardware_interface_type_values.hpp"
 #include "lifecycle_msgs/msg/state.hpp"
 #include "pluginlib/class_loader.hpp"
 #include "rcutils/logging_macros.h"
 
+#define COMPONENT_NAME_COMPARE [&](const auto & name) { return name == component.get_name(); }
```

Comment:
in 2022 I don't want to accept code that does `#define` for functions, sorry

Reply:
To bad we didin't merge this in 2021 :D

Ok, I had the issue that I could not define/use lambda inside a lambda function. So, I opted out to this solution. 

Nevertheless, it seems this is not needed, so I am removing it completely.